src/components/shared/SpentTime/SpentTime.tsx

```diff
@@ -0,0 +1,128 @@
+import React, { useCallback, useEffect, useMemo, useState } from "react";
+import { useFirestore } from "reactfire";
+import { collection, getDocs, query, where } from "firebase/firestore";
+import { chunk } from "lodash";
+
+import {
+  COLLECTION_SPACES,
+  FIRESTORE_QUERY_IN_ARRAY_MAX_ITEMS,
+} from "settings";
+
+import { getUserRef } from "api/profile";
+
+import { UserVisit } from "types/Firestore";
+import { UserId } from "types/id";
+import { AnyVenue } from "types/venues";
+
+import { WithId } from "utils/id";
+import { isTruthy } from "utils/types";
+
+import { useWorldAndSpaceByParams } from "hooks/spaces/useWorldAndSpaceByParams";
+
+import styles from "./SpentTime.module.scss";
+
+interface SpentTimeProps {
+  userId: UserId;
+}
+
+export const SpentTime: React.FC<SpentTimeProps> = ({ userId }) => {
+  const { isLoaded: isWorldLoaded, spaceId } = useWorldAndSpaceByParams();
+  const [visits, setVisits] = useState<WithId<UserVisit>[]>([]);
+  const [venues, setVenues] = useState<WithId<AnyVenue>[]>([]);
+  const [isLoading, setIsLoading] = useState<boolean>(false);
+
+  const firestore = useFirestore();
```

Comment:
All this firestore code looks like it doesn't belong here. At least it can use a `@debt` comment to be moved elsewhere.

I would really like it if Reactfire isn't used new new code, since it's on its way to be removed.

There is a `getFirestore()` function, a `FIREBASE.firestore` singleton as well as `useFire*` and `useLive*` hooks to pick from.

Reply:
I was hoping to at least have that `useFirestore()` replaced by `getFirestore()` so there isn't an import from `Reactfire`.

Reply:
done: https://github.com/sparkletown/sparkle/pull/2953/commits/d110155d478fa8fcec0c3dce7443ed2d4b871a01